File: computePolkadotScore.java
public final class AngelicaPolkadotScore {
    private AngelicaPolkadotScore() {}

    public static int computePolkadotScore(String asciiArt) {
        String[] lines = asciiArt.split("\\R", -1);

        int eyeLineIndex = -1;
        for (int i = 0; i < (lines.length + 1) / 2; i++) {
            int bullets = 0;
            for (int j = 0; j < lines[i].length(); j++) {
                if (lines[i].charAt(j) == '•') {
                    bullets++;
                }
            }
            if (bullets >= 2) {
                eyeLineIndex = i;
                break;
            }
        }

        if (eyeLineIndex == -1) {
            throw new IllegalArgumentException("Could not find Angelica's pupils.");
        }

        int pupilCharCount = 0;
        for (int j = 0; j < lines[eyeLineIndex].length(); j++) {
            if (lines[eyeLineIndex].charAt(j) == '•') {
                pupilCharCount++;
            }
        }

        if (eyeLineIndex + 1 >= lines.length) {
            throw new IllegalArgumentException("Could not find Angelica's lips.");
        }

        String lipLine = lines[eyeLineIndex + 1];
        int lipStart = -1;
        int lipEnd = -1;
        for (int j = 0; j < lipLine.length(); j++) {
            if (lipLine.charAt(j) == ';') {
                if (lipStart == -1) {
                    lipStart = j;
                }
                lipEnd = j;
            }
        }

        if (lipStart == -1 || lipEnd == -1) {
            throw new IllegalArgumentException("Could not determine the lips range.");
        }

        int inside = 0;
        int outside = 0;

        for (String line : lines) {
            for (int j = 0; j < line.length(); j++) {
                if (line.charAt(j) == 'O') {
                    if (j >= lipStart && j <= lipEnd) {
                        inside++;
                    } else {
                        outside++;
                    }
                }
            }
        }

        return outside + inside * pupilCharCount;
    }

    public static void main(String[] args) {
        String art = """
                      ,   ,-',
                ,', ,'  ','  ,'   ÅÑGË£ÏÇÄ †(–)Ë ßRÄ†
              '-',  '      ,'
                  ' -,    ',
                      ' -, ',                                         , - - -,
                     ('''''' ®'''''''')                        ,,,,,    ,-' -,''''''''',
                      ` ~„''`„~ '                          ',  ,', -' , -,'' '''''''''',
                          "„  " - „                   „ - ",®,-'     `~~' '''''''
                         „"         " „         „ - "      ,',,,',
                       „-" " " " " " " ~~~~~~" - „       ,'
                    „" –,'' ~ ,       • ; •          "      "„
                    """";      ' - , ,  ; , , , - '' ' ' -,_ ', ',
                   , -' ' ',           ,'    ,'             ',~', ',
                 ,'         ' - , ,()' /\\    ',          (),'¯ ,'   `¸`;
                 ',                            ` ` ` ` ` `      ,-,,,-'
                   '-,                                  ,¬  ,-'
                      ' -, ~            ~~~~~~' ' ` ,-'
                          `~-,,,,,,,      ,,,,,,,,,,-~'
              ('('('(,,,              ;    ;                •Å(V)åö•
               '-, '-,'''      ,-';`,`'ˆˆˆˆˆ ,' ;' ' -,           •97•
                 ;¯ ;      ;  ;  ', ; ; ,'  ;  ‚¸  ' -,        •••
                 ;   ;     ;       '''''''''''    ``'-,',  ,'
                 ;   ;, -¬;    O   O   O  O   '-',,,,,,,,,,,,
                 ;        ;  O   O     O O    O  ,'     O     ,'
                  ' - - ' `;    O        O  O    O   O    ,'
                       ,-'   O    O    O  O      O    O   ,'
                    ,-' O O  O   O        O        O ,'
                 ,-'   O      O   O   O     O  O     ,'
              ,-'  O    O    O  O   O   O O O   O,-'-,
               ``¬ -,,,,,,,-¬~,~~~~~~~~~--',)  (' -,
                            ',   (',                     ' -,    '-,
                             ',)  (',                        `-,)  ' -,
                              ',    ',                           `-,  ,',-----,
                               ',)   ;                              `\\,- ---'
                           ¸,,,,'‡  (;
                          (¸,,,,,';_'\\ ßy §(V)òó†(–)775 ™
                """;

        System.out.println(computePolkadotScore(art)); // 64
    }
}

File: computePolkadotScore.js
"use strict";

function computePolkadotScore(asciiArt) {
  const lines = asciiArt.split(/\r?\n/);

  const eyeLineIndex = lines.findIndex(
    (line, index) => index < Math.ceil(lines.length / 2) && [...line].filter((ch) => ch === "•").length >= 2
  );

  if (eyeLineIndex === -1) {
    throw new Error("Could not find Angelica's pupils.");
  }

  const pupilCharCount = [...lines[eyeLineIndex]].filter((ch) => ch === "•").length;
  const lipLine = lines[eyeLineIndex + 1];

  if (!lipLine) {
    throw new Error("Could not find Angelica's lips.");
  }

  // The next line contains the smile corners as semicolons; the special
  // range spans from the first lip character to the last lip character.
  const lipColumns = [];
  [...lipLine].forEach((ch, column) => {
    if (ch === ";") {
      lipColumns.push(column);
    }
  });

  if (lipColumns.length < 2) {
    throw new Error("Could not determine the lips range.");
  }

  const lipStart = Math.min(...lipColumns);
  const lipEnd = Math.max(...lipColumns);

  let inside = 0;
  let outside = 0;

  for (const line of lines) {
    [...line].forEach((ch, column) => {
      if (ch !== "O") {
        return;
      }

      if (column >= lipStart && column <= lipEnd) {
        inside += 1;
      } else {
        outside += 1;
      }
    });
  }

  return outside + inside * pupilCharCount;
}

module.exports = { computePolkadotScore };

if (require.main === module) {
  const fs = require("node:fs");
  const path = process.argv[2] || "angelica.txt";
  const art = fs.readFileSync(path, "utf8");

  console.log(computePolkadotScore(art));
}
